### DISCUSSION TITLE
Blue Onyx 0.1.0

### DISCUSSION THREAD

**Discussion by xnorpx** [xnorpx/blue-onyx#2] [Announcements]
## Welcome to Blue Onyx  

**Blue Onyx** is an object detection service that processes JPG images, identifies objects, and returns bounding boxes for the detected items. It is fully API-compatible with both **Blue Iris** and **Agent DVR**.  

Blue Onyx is a streamlined refactor of [[Blue Candle](https://github.com/xnorpx/blue-candle)](https://github.com/xnorpx/blue-candle), with several key improvements and changes.  

### Key Differences Between Blue Candle and Blue Onyx:  
- **(Still) No Coral Support**  
- **Intel and AMD GPU Support**  
- **Custom Model Support**: You can use your own trained models.  
- **Inference Backend**: Switched from Candle to ONNX.  
- **Model Update**: Replaced YOLOv8 with [RT-DETR](https://github.com/lyuwenyu/RT-DETR)v2 ([RT-DETR](https://github.com/lyuwenyu/RT-DETR)).  
- **DirectML on Windows**: Uses DirectML instead of CUDA for easier builds and broader support.  
- **Benchmark Tool**: Added a binary to run object detection outside of the service.  
- **Windows-Only Support**: No Linux builds are available.  
- **Simpler Architecture**: Improved design for ease of use.  
- **CLI Updates**: Some command-line interface changes.  

### Additional Information  
More FAQs and detailed documentation will be provided soon.  

---

![Blue Onyx Interface](https://github.com/user-attachments/assets/bb5d2a1f-7845-45d2-979b-3aea65b66fb7)

---

**Comment by slim-bean** [comment]
WOOOOOO!!!! So excited to see you continuing to work on this, I discovered Blue Candle a few months ago and was thrilled with just how much better the experience both in operation and performance is running it vs deepstack or codeproject.ai

Really my only nit was the lack of custom models and some trouble I was having with detecting vehicles at night.

Very excited to see more work being done here, thank you so much for sharing this!

I think you should enroll into the [Github Sponsors](https://docs.github.com/en/sponsors) program (or something similar) to allow folks with the means to compensate you for your time and work!

❤️
